### PR TITLE
Change fields from long to Duration(#111)

### PR DESCRIPTION
**Pull request by ShigapovRinat**

### CODE REVIEW

File: roc_jni/src/main/impl/receiver.c [hunk 1]
```diff
@@ -54,35 +54,59 @@ static int receiver_config_unmarshal(JNIEnv* env, roc_receiver_config* config, j
     if (jobj != NULL)
         config->resampler_profile = (roc_resampler_profile) get_resampler_profile(env, jobj);
 
+    durationClass = (*env)->FindClass(env, "java/time/Duration");
+    assert(protocolClass != NULL);
```
Comment (ortex) [inline roc_jni/src/main/impl/receiver.c:58]: protocolClass -> durationClass

also you need to declare a variable for `durationClass`.

File: roc_jni/src/main/impl/receiver.c [hunk 2]
```diff
@@ -54,35 +54,59 @@ static int receiver_config_unmarshal(JNIEnv* env, roc_receiver_config* config, j
...
+            = (*env)->GetStaticMethodID(env, durationClass, "ofNanos", "(J)Ljava/time/Duration;");
+    assert(getOfNanosMethodID != NULL);
+
     // target_latency
-    config->target_latency
+    targetLatencyLong
         = get_ullong_field_value(env, receiverConfigClass, jconfig, "targetLatency", &err);
     if (err) return err;
+    config->target_latency = (jlong) (*env)->CallStaticObjectMethod(
+        env, durationClass, getOfNanosMethodID, (jlong) targetLatencyLong);
```
Comment (ortex) [inline roc_jni/src/main/impl/receiver.c:68]: here need to create from java-config C config. Looks like you do the opposite

File: src/main/java/org/rocstreaming/roctoolkit/Check.java [hunk 3]
```diff
@@ -25,8 +27,8 @@ static int notNegative(int value, String name) {
         return value;
     }
 
-    static long notNegative(long value, String name) {
-        if (value < 0) {
+    static Duration notNegative(Duration value, String name) {
```
Comment (ortex) [inline src/main/java/org/rocstreaming/roctoolkit/Check.java:30]: value could be null here. I guess need to rename it to `notNegativeOrNull` and check if value is null

File: roc_jni/src/main/impl/receiver.c (hunk 2 above)
Comment (ortex) [reply roc_jni/src/main/impl/receiver.c:68]: also keep in mind that Duration fields in java config could be null. No need to set it to C config in such case

File: roc_jni/src/main/impl/receiver.c (hunk 1 above)
Comment (ShigapovRinat) [reply roc_jni/src/main/impl/receiver.c]: fix

File: roc_jni/src/main/impl/receiver.c (hunk 2 above)
Comment (ShigapovRinat) [reply roc_jni/src/main/impl/receiver.c]: change

File: roc_jni/src/main/impl/duration.c [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#include "duration.h"
+#include "common.h"
+
+#include <roc/config.h>
+
+roc_duration get_duration(JNIEnv* env, jobject jduration) {
```
Comment (ortex) [inline roc_jni/src/main/impl/duration.c:6]: what is `roc_duration`?

File: src/main/java/org/rocstreaming/roctoolkit/Check.java [hunk 5]
```diff
@@ -25,8 +27,9 @@ static int notNegative(int value, String name) {
         return value;
     }
 
-    static long notNegative(long value, String name) {
-        if (value < 0) {
+    static Duration notNegativeOrNull(Duration value, String name) {
```
Comment (ortex) [inline src/main/java/org/rocstreaming/roctoolkit/Check.java:30]: `notNegativeOrNull` - name suggests that it allows value to be null

**Comment by gavv** [comment]
@ShigapovRinat Hi, do you still have plans on this?

**Comment by gavv** [comment]
Closing, feel free to reopen if you decide to return to this

**Pull request closed without merging**